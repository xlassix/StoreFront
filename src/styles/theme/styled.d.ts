import 'styled-components';
import {
  Theme as ColorTheme,
  Direction,
  BorderStatus,
  ShadowStatus,
} from './themes';

export type Theme = {
  color: ColorTheme;
  direction: Direction;
  shadow: ShadowStatus;
  border: BorderStatus;
};

export type ThemeClasses = {
  color: 'theme-dark' | 'theme-light';
  direction: 'ltr' | 'rtl';
  shadow: 'blocks-shadows-on' | 'blocks-shadows-off';
  border: 'round-borders-on' | 'round-borders-off';
};
/* This is the suggested way of declaring theme types */
declare module 'styled-components' {
  export interface DefaultTheme extends Theme {}
}
