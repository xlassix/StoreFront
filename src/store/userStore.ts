import { action, thunk, createStore, Thunk, Action } from 'easy-peasy';
import { User } from 'types/User';
import { EntityStore } from 'types/EntityStore';

interface UserModel {
  user: {};
  selectedStore: {};
  search: string | null;
  setUser: Action<UserModel, User>;
  setSelectStore: Action<UserModel, EntityStore>;
  setSearch: Action<UserModel, string>;
}

export const userStore = createStore<UserModel>({
  user: {},
  selectedStore: {},
  search: '',
  setUser: action((state, payload: User) => {
    state.user = { ...payload, isAuthenticated: true };
  }),
  setSelectStore: action((state, payload: EntityStore) => {
    state.selectedStore = payload;
  }),
  setSearch: action((state, payload: string) => {
    state.search = payload;
  }),
});
