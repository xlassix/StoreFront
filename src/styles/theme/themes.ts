import lightLogo from '../../app/layouts/components/Topbar/assets/bs.gif';
import darkLogo from '../../app/layouts/components/Topbar/assets/bs.gif';

const lightTheme = {
  colorBackground: 'white',
  colorBackgroundBody: '#f2f4f7',
  colorHeaderDataPicker: '#6db0ff',
  colorText: '#646777',
  colorTextAdditional: '#646777',
  colorHover: '#fafbfe',
  colorFolderHover: '#f0eeee',
  colorBorder: '#eff1f5',
  colorIcon: '#dddddd',
  colorFieldsBorder: '#f2f4f7',
  colorBubble: 'rgba(242, 244, 247, 0.65)',
  colorBubbleActive: 'rgba(234, 238, 255, 0.6)',
  colorScrollbar: '#B4BFD0',
  colorFitness: '#646777',
  colorEmoji: '#232329',
  inboxButtonBackground: '#f2f4f7',
  inboxButtonBackgroundHover: 'rgba(242, 244, 247, 0.05)',
  sidebarColor: 'aliceblue',
  logoImg: `url(${darkLogo})`,
};

const darkTheme: Theme = {
  colorBackground: '#232329',
  colorBackgroundBody: '#2a2a31',
  colorHeaderDataPicker: '#063263',
  colorText: '#dddddd',
  colorTextAdditional: '#999999',
  colorHover: '#38373f',
  colorFolderHover: '#ffffff1A',
  colorBorder: '#333246',
  colorIcon: '#605f7b',
  colorFieldsBorder: '#33333a',
  colorBubble: 'rgba(68, 79, 97, 0.65)',
  colorBubbleActive: 'rgba(92, 104, 156, 0.6)',
  colorScrollbar: '#606071',
  colorFitness: '#ffffff',
  colorEmoji: '#ffffff',
  inboxButtonBackground: '#2a2a31',
  inboxButtonBackgroundHover: 'rgba(42, 42, 49, 0.05)',
  sidebarColor: '#232329',
  logoImg: `url(${lightLogo})`,
};

const ltr = {
  direction: 'ltr',

  right: 'right',
  left: 'left',

  'margin-right': 'margin-right',
  'margin-left': 'margin-left',

  'padding-left': 'padding-left',
  'padding-right': 'padding-right',

  'border-top-left-radius': 'border-top-left-radius',
  'border-bottom-left-radius': 'border-bottom-left-radius',

  'border-top-right-radius': 'border-top-right-radius',
  'border-bottom-right-radius': 'border-bottom-right-radius',

  'border-left': 'border-left',
  'border-right': 'border-right',

  translate: 'translate(-50%, -50%);',
  mirrorY: 'scale(1, 1)',

  'border-right-color': 'border-right-color',

  'transform-sidebar': 'rotate(90deg)',

  'flex-flow': 'row nowrap',
  row: 'row',

  'sidebar-close': 'translateX(0)',
  'sidebar-no-desktop': 'translateX(calc(0%))',
  '-': '-',
};

const rtl: Direction = {
  direction: 'rtl',

  right: 'left',
  left: 'right',

  'margin-right': 'margin-left',
  'margin-left': 'margin-right',

  'padding-left': 'padding-right',
  'padding-right': 'padding-left',

  'border-top-left-radius': 'border-top-right-radius',
  'border-bottom-left-radius': 'border-bottom-right-radius',

  'border-top-right-radius': 'border-top-left-radius',
  'border-bottom-right-radius': 'border-bottom-left-radius',

  'border-left': 'border-right',
  'border-right': 'border-left',

  translate: 'translate(50%, -50%);',
  mirrorY: 'scale(1, -1)',

  'border-right-color': 'border-left-color',
  'transform-sidebar': 'rotate(-90deg)',

  'flex-flow': 'row-reverse nowrap',
  row: 'row-reverse',

  'sidebar-close': 'translateX(0)',
  'sidebar-no-desktop': 'translateX(calc(100%))',
  '-': '+',
};

const roundBordersOn = {
  borderRadius: '15px',
};

const roundBordersOff: BorderStatus = {
  borderRadius: '5px',
};

const shadowOn = {
  blocksShadows: '0 10px 30px 1px rgba(0, 0, 0, 0.06)',
};
const shadowOff: ShadowStatus = {
  blocksShadows: 'none',
};

export type Theme = typeof lightTheme;
export type Direction = typeof ltr;
export type BorderStatus = typeof roundBordersOn;
export type ShadowStatus = typeof shadowOn;
export const themes = {
  colors: {
    light: lightTheme,
    dark: darkTheme,
  },
  directions: {
    ltr: ltr,
    rtl: rtl,
  },
  borderStatus: {
    on: roundBordersOn,
    off: roundBordersOff,
  },
  shadowStatus: {
    on: shadowOn,
    off: shadowOff,
  },
};
