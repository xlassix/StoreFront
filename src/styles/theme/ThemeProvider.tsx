import * as React from 'react';
import { ThemeProvider as OriginalThemeProvider } from 'styled-components';
import { useSelector } from 'react-redux';
import { useThemeSlice } from './slice';
import {
  selectTheme,
  selectDirection,
  selectBorder,
  selectShadow,
} from './slice/selectors';

export const ThemeProvider = (props: { children: React.ReactChild }) => {
  useThemeSlice();

  const colorTheme = useSelector(selectTheme);
  const directionTheme = useSelector(selectDirection);
  const borderTheme = useSelector(selectBorder);
  const shadowTheme = useSelector(selectShadow);

  return (
    <OriginalThemeProvider
      theme={{
        color: colorTheme,
        direction: directionTheme,
        shadow: shadowTheme,
        border: borderTheme,
      }}
    >
      {React.Children.only(props.children)}
    </OriginalThemeProvider>
  );
};
