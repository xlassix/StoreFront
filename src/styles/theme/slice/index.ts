import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { useInjectReducer } from 'utils/redux-injectors';
import { getThemeFromStorage } from '../utils';
import {
  ThemeKeyType,
  DirectionKeyType,
  BorderStatusKeyType,
  ShadowStatusKeyType,
  ThemeState,
} from './types';

export const initialState: ThemeState = {
  selected: getThemeFromStorage() || 'system',
  selectedDir: 'ltr',
  selectedBorderStatus: 'on',
  selectedShadowStatus: 'on',
};

const slice = createSlice({
  name: 'theme',
  initialState,
  reducers: {
    changeTheme(state, action: PayloadAction<ThemeKeyType>) {
      state.selected = action.payload;
    },
    changeDirection(state, action: PayloadAction<DirectionKeyType>) {
      state.selectedDir = action.payload;
    },
    changeBorder(state, action: PayloadAction<BorderStatusKeyType>) {
      state.selectedBorderStatus = action.payload;
    },
    changeShadow(state, action: PayloadAction<ShadowStatusKeyType>) {
      state.selectedShadowStatus = action.payload;
    },
  },
});

export const { actions: themeActions, reducer } = slice;

export const useThemeSlice = () => {
  useInjectReducer({ key: slice.name, reducer: slice.reducer });
  return { actions: slice.actions };
};
