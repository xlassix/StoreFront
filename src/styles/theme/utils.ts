import { ThemeKeyType, ThemeState } from './slice/types';
import { ThemeClasses } from './styled';

/* istanbul ignore next line */
export const isSystemDark = window?.matchMedia
  ? window.matchMedia('(prefers-color-scheme: dark)')?.matches
  : undefined;

export function saveTheme(theme: ThemeKeyType) {
  window.localStorage && localStorage.setItem('selectedTheme', theme);
}

/* istanbul ignore next line */
export function getThemeFromStorage(): ThemeKeyType | null {
  return window.localStorage
    ? (localStorage.getItem('selectedTheme') as ThemeKeyType) || null
    : null;
}

/**
 *
 * @name convertToThemeClass
 *
 * @description This function converts the keys of the themeState objects to classNames required by styling
 *
 * @return ThemeClasses
 */
export const convertThemeToClass = (theme: ThemeState): ThemeClasses => {
  let color: ThemeClasses['color'];

  if (theme.selected === 'system') {
    color = isSystemDark ? 'theme-dark' : 'theme-light';
  } else {
    color = theme.selected === 'dark' ? 'theme-dark' : 'theme-light';
  }

  const direction = theme.selectedDir === 'ltr' ? 'ltr' : 'rtl';
  const border =
    theme.selectedBorderStatus === 'on'
      ? 'round-borders-on'
      : 'round-borders-off';
  const shadow =
    theme.selectedShadowStatus === 'on'
      ? 'blocks-shadows-on'
      : 'blocks-shadows-off';

  return {
    color,
    direction,
    border,
    shadow,
  };
};
