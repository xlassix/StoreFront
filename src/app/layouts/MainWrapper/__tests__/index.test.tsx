import * as React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { Store } from '@reduxjs/toolkit';

import { ThemeProvider } from 'styles/theme/ThemeProvider';
import { MainWrapper } from '..';
import { configureAppStore } from 'store/configureStore';

const renderWrapper = (store: Store) =>
  render(
    <Provider store={store}>
      <ThemeProvider>
        <MainWrapper>
          <div></div>
        </MainWrapper>
      </ThemeProvider>
    </Provider>,
  );

describe.only('<MainWrapper  />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    store = configureAppStore();
  });

  it('should match snapshot', () => {
    const wrapper = renderWrapper(store);
    expect(wrapper.container.firstChild).toMatchSnapshot();
  });
});
