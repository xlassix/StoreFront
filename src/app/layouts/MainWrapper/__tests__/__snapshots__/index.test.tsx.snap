// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainWrapper  /> should match snapshot 1`] = `
<div
  class="theme-light 
              round-borders-on
              blocks-shadows-on
              ltr-support"
  dir="ltr"
>
  <div
    class="wrapper top-navigation"
  >
    <div />
  </div>
</div>
`;
